docs/source/dev/how-io-works.rst

```diff
@@ -0,0 +1,134 @@
+How does I/O work in LiberTEM?
+==============================
+
```

Comment:
Many algorithms benefit from :ref:`tiled` where the same subset in the signal dimension is processed for several frames in a row. In many cases, algorithms have specific minimum and maximum sizes in signal dimension, navigation dimension or total size where they operate efficiently. Smaller sizes might increase overheads, while larger sizes might reduce cache efficiency.

On the other side, file formats might operate well within specific size and shape limits. The :ref:`k2is` raw format is a prime example where data is saved in tiled form and can be processed efficiently in specific tile sizes and shapes that follow the native layout. Furthermore, some formats require decoding or corrections by the CPU, such as :ref:`frms6`, where tiles that fit the L3 cache can speed up subsequent processing steps. Requirements from the I/O method such as alignment and efficient block sizes are taken into account as well.

The LiberTEM I/O back-end negotiates a tiling scheme between UDF and dataset that fulfills requirements from both UDF and dataset side as far as possible. However, it is not always guaranteed that the supplied data will fall within the requested limits.